unityfiles/Assets/Scripts/Audio/AudioManager.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlaySoundEventArgs : EventArgs {
+}
+
+public class AudioManager : Singleton<AudioManager> {
+    
+    public Sound[] sounds;
+    
+    private void Awake() {
+        SubscribeToEvents();
+        
+        DontDestroyOnLoad(gameObject);
+        
+        foreach (Sound s in sounds) {
+            s.audioSource = gameObject.AddComponent<AudioSource>();
+            s.audioSource.clip = s.AudioClip;
```

Comment:
s.audioSource.clip  er null her og skaper null pointer litt lengere ned

Reply:
dette er forde du ikke har fylt in alle elementene i sounds arrayet, s\ n\r den looper gjennom alle elementene og treffer de som er null ender man opp med en del null pointere